File: frontend/src/services/compensationService.js
import apiClient from './api';

export const getBands = () => apiClient.get('/compensation/bands');
export const addBand = (dto) => apiClient.post('/compensation/bands', dto);
export const getSalaryReviews = () => apiClient.get('/compensation/reviews');
export const submitReview = (dto) => apiClient.post('/compensation/reviews', dto);
export const actionReview = (id, status) => apiClient.post(`/compensation/reviews/${id}/action`, { status });

File: backend/src/main/java/com/awais/hr/module/compensation/service/CompensationService.java
package com.awais.hr.module.compensation.service;

import java.util.List;
import java.util.Map;

public interface CompensationService {
    List<Map<String, Object>> getBands();
    void addBand(Map<String, Object> body);
    List<Map<String, Object>> getSalaryReviews();
    void submitReview(String requesterEmail, Map<String, Object> body);
    void actionReview(String reviewerEmail, String reviewId, String status);
}
